robot/Cumulus/resources/NPSP.py

```diff
@@ -1288,4 +1280,57 @@ def verify_toast_message(self,value):
                 found=True
                 print("Toast message verified")
                 break
-        assert found, "Expected Toast message not found on page"
+        assert found, "Expected Toast message {} not found on page".format(value)
+
+    def edit_record_field_value(self,field,value):
+        """Scrolls just a little below the field
+           Clicks on Edit icon next to field and enters a value into the field"""
+        scroll_loc=npsp_lex_locators["span_button"].format(field)
+        # To make sure the field we want to edit has rendered
+        # and is not obscured by the footer, scroll down a little below the element
+        self.selenium.scroll_element_into_view(scroll_loc)
+        self.selenium.execute_javascript("window.scrollBy(0,50)")
+        btn="Edit "+field
+        self.selenium.click_button(btn)
+        footer=npsp_lex_locators["record"]["footer"]
+        self.selenium.wait_until_page_contains_element(footer)
+        self.salesforce.populate_lookup_field(field,value)
+    
+    def edit_record_checkbox(self,field,status):
+        """Scrolls just a little below the field
+           Clicks on Edit icon next to field
+           checks if status is 'checked' 
+           unchecks if status in 'unchecked'"""
+        scroll_loc=npsp_lex_locators["span_button"].format(field)
+        # To make sure the field we want to edit has rendered
+        # and is not obscured by the footer, scroll down a little below the element
+        self.selenium.scroll_element_into_view(scroll_loc)
+        self.selenium.execute_javascript("window.scrollBy(0,50)")
```

Comment:
Why do you have to scroll and execute javascript? Is the view of the element blocked by something ?

Reply:
yes we have to scroll just a little below the actual element we want to click otherwise click is obstructed by footer